### PR TITLE
Cleanup z.fs

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
A wild @forki appears!

**Review by KevinRansom — CHANGES_REQUESTED** [review]

**Comment by KevinRansom** [comment]
@vzarytovskii , @TIHan :

File: src/fsharp/FSharp.Core/math/z.fs [hunk 1]
```diff
@@ -40,16 +38,16 @@ namespace Microsoft.FSharp.Core
                     tab64.[value] <- res
                     res)                 
 
-            let inline get32 (x32:int32) =  FromInt64Dynamic (int64 x32)
+            let inline get32 (x32:int32) = FromInt64Dynamic (int64 x32)
```
[the remark this answers is not in the document]
Comment (forki) [reply src/fsharp/FSharp.Core/math/z.fs]: done

**Comment by vzarytovskii** [comment]
> @vzarytovskii , @TIHan : Some new threading issue?
> 
> Failure:
> 
> ```
> The active test run was aborted. Reason: Test host process crashed : Unhandled exception. System.ObjectDisposedException: Safe handle has been closed.
> Object name: 'SafeHandle'.
>    at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
>    at System.Runtime.InteropServices.SafeBuffer.AcquirePointer(Byte*& pointer)
>    at System.IO.MemoryMappedFiles.MemoryMappedView.Flush(UIntPtr capacity)
>    at System.IO.MemoryMappedFiles.MemoryMappedViewStream.Dispose(Boolean disposing)
>    at System.IO.Stream.Close()
> ```

Should be fixed in `main`

**Comment by forki** [comment]
@vzarytovskii I think I rebased this morning. Want me to rebase it again?

**Comment by vzarytovskii** [comment]
> @vzarytovskii I think I rebased this morning. Want me to rebase it again?

Yeah, could you rebase again please, I don't see the fix in your branch, I probably merged it after you've rebased.

**Comment by forki** [comment]
still red after yet another rebase

**Comment by vzarytovskii** [comment]
Hm, I don't get it, i've fixed this particular issue in the main.

**Comment by vzarytovskii** [comment]
Hm, I see some weird test results, for example F#QA:
```
Peverify: [D:\workspace\_work\1\s\tests\fsharpqa\source\..\..\..\artifacts\bin\PEVerify\Release\net472\PEVerify.exe parse_tests.dll ]
[D:\workspace\_work\1\s\tests\fsharpqa\source\Libraries\Portable\parse_tests.dll : FSLib+BigIntTest::T][mdToken=0x6000001][offset 0x00000003][found ref 'System.Object'][expected value 'System.Numerics.BigInteger'] Unexpected type on the stack.

[D:\workspace\_work\1\s\tests\fsharpqa\source\Libraries\Portable\parse_tests.dll : FSLib+BigIntTest::T][mdToken=0x6000001][offset 0x00000008][found ref 'System.Object'][expected value 'System.Numerics.BigInteger'] Unexpected type on the stack.
PeVerify Failed the test
```
And
```
[xUnit.net 00:02:41.75]     FSharp.Core.UnitTests.Math.BigIntType.ObjectToString [FAIL]
  Failed FSharp.Core.UnitTests.Math.BigIntType.Equality [< 1 ms]
  Error Message:
   System.Exception : Assertion: Expected True but got False
  Stack Trace:
     at FSharp.Core.UnitTests.Math.BigIntType.Equality() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 356
  Failed FSharp.Core.UnitTests.Math.BigIntType.Parse [< 1 ms]
  Error Message:
   System.FormatException : The value is too large to be represented by this format specifier.
---- System.OverflowException : Arithmetic operation resulted in an overflow.
  Stack Trace:
     at System.Numerics.BigNumber.FormatBigInteger(Boolean targetSpan, BigInteger value, String formatString, ReadOnlySpan`1 formatSpan, NumberFormatInfo info, Span`1 destination, Int32& charsWritten, Boolean& spanSuccess)
   at System.Numerics.BigNumber.FormatBigInteger(BigInteger value, String format, NumberFormatInfo info)
   at System.Numerics.BigInteger.ToString()
   at FSharp.Core.UnitTests.Math.BigIntType.Parse() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 201
----- Inner Stack Trace -----
   at System.Numerics.BigNumber.FormatBigInteger(Boolean targetSpan, BigInteger value, String formatString, ReadOnlySpan`1 formatSpan, NumberFormatInfo info, Span`1 destination, Int32& charsWritten, Boolean& spanSuccess)
  Failed FSharp.Core.UnitTests.Math.BigIntType.GreaterThan [< 1 ms]
  Error Message:
   System.Exception : Assertion: Expected True but got False
  Stack Trace:
     at FSharp.Core.UnitTests.Math.BigIntType.GreaterThan() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 371
  Failed FSharp.Core.UnitTests.Math.BigIntType.LessThanOrEqual [< 1 ms]
  Error Message:
   System.Exception : Assertion: Expected True but got False
  Stack Trace:
     at FSharp.Core.UnitTests.Math.BigIntType.LessThanOrEqual() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 420
  Failed FSharp.Core.UnitTests.Math.BigIntType.ObjectToString [< 1 ms]
  Error Message:
   System.FormatException : The value is too large to be represented by this format specifier.
---- System.OverflowException : Arithmetic operation resulted in an overflow.
  Stack Trace:
     at System.Numerics.BigNumber.FormatBigInteger(Boolean targetSpan, BigInteger value, String formatString, ReadOnlySpan`1 formatSpan, NumberFormatInfo info, Span`1 destination, Int32& charsWritten, Boolean& spanSuccess)
   at System.Numerics.BigNumber.FormatBigInteger(BigInteger value, String format, NumberFormatInfo info)
   at System.Numerics.BigInteger.ToString()
   at FSharp.Core.UnitTests.Math.BigIntType.ObjectToString() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 44
----- Inner Stack Trace -----
   at System.Numerics.BigNumber.FormatBigInteger(Boolean targetSpan, BigInteger value, String formatString, ReadOnlySpan`1 formatSpan, NumberFormatInfo info, Span`1 destination, Int32& charsWritten, Boolean& spanSuccess)
[xUnit.net 00:02:41.75]     FSharp.Core.UnitTests.Math.BigIntType.GreaterThanOrEqual [FAIL]
[xUnit.net 00:02:41.76]     FSharp.Core.UnitTests.Math.BigIntType.Sign [FAIL]
  Failed FSharp.Core.UnitTests.Math.BigIntType.GreaterThanOrEqual [< 1 ms]
  Error Message:
   System.Exception : Assertion: Expected True but got False
  Stack Trace:
     at FSharp.Core.UnitTests.Math.BigIntType.GreaterThanOrEqual() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 385
  Failed FSharp.Core.UnitTests.Math.BigIntType.Sign [< 1 ms]
  Error Message:
   System.Exception : Assertion: Expected 0 but got 1
  Stack Trace:
     at FSharp.Core.UnitTests.Math.BigIntType.Sign() in /Users/runner/work/1/s/tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Core/BigIntType.fs:line 241
```

It doesn't seem to be related to the FileSystem issues we've had.

**Comment by forki** [comment]
never mind.

**Pull request closed without merging**